dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/processor/TaskKillProcessor.java

```diff
@@ -129,11 +127,19 @@ public void process(Channel channel, Command command) {
 
             OSUtils.exeCmd(cmd);
 
-            // find log and kill yarn job
-            appIds = killYarnJob(Host.of(taskExecutionContext.getHost()).getIp(),
-                taskExecutionContext.getLogPath(),
-                taskExecutionContext.getExecutePath(),
-                taskExecutionContext.getTenantCode());
+            TaskType anEnum = EnumUtils.getEnum(TaskType.class, taskExecutionContext.getTaskType());
+            switch (anEnum) {
+                case SPARK:
+                case FLINK:
+                    // find log and kill yarn job
+                    appIds = killYarnJob(Host.of(taskExecutionContext.getHost()).getIp(),
+                            taskExecutionContext.getLogPath(),
+                            taskExecutionContext.getExecutePath(),
+                            taskExecutionContext.getTenantCode());
+                    break;
+                default:
+                    appIds = Collections.EMPTY_LIST;
+            }
```

Comment:
hive-sql,mr,YARN tasks started in shell/pythod scripts, and so on. These all involve killing yarn tasks, and this change is not recommended here

Reply:
not all job is run on yarn cluster, 
this place need optimize